### ISSUE TITLE
Compressing certain files results in corrupted ZIP file

### ISSUE THREAD

**Issue by chenxiaolong** [nmoinvaz/minizip#218]
Compressing certain files seems to result in corrupted ZIP files. I haven't figured out what causes this as it only seems to occur with one file out of ~2000 that I'm adding to the zip.

For example, if I try to add the following file to a ZIP using the `minizip` test executable: https://noobdev.io/github/minizip/218/mbtool

I get the following:
```
[build] rm /tmp/test.zip
[build] ./minizip /tmp/test.zip /tmp/mbtool
Minizip 2.2.4 - https://github.com/nmoinvaz/minizip
---------------------------------------------------
Creating /tmp/test.zip
Adding: tmp/mbtool
[build] unzip -t /tmp/test.zip
Archive:  /tmp/test.zip
    testing: tmp/mbtool              
  error:  invalid compressed data to inflate
At least one error was detected in /tmp/test.zip.
```

7zip reports the same issue with the file as well. This is the output zip that was created: https://noobdev.io/github/minizip/218/test.zip

---

This is what `zipinfo -v` says about the file:
```
Archive:  /tmp/test.zip
There is no zipfile comment.

End-of-central-directory record:
-------------------------------

  Zip archive file size:                    655538 (00000000000A00B2h)
  Actual end-cent-dir record offset:        655516 (00000000000A009Ch)
  Expected end-cent-dir record offset:      655516 (00000000000A009Ch)
  (based on the length of the central directory and its expected offset)

  This zipfile constitutes the sole disk of a single-part archive; its
  central directory contains 1 entry.
  The central directory is 84 (0000000000000054h) bytes long,
  and its (expected) offset in bytes from the beginning of the zipfile
  is 655432 (00000000000A0048h).


Central directory entry #1:
---------------------------

  tmp/mbtool

  offset of local header from start of archive:   0
                                                  (0000000000000000h) bytes
  file system or operating system of origin:      MS-DOS, OS/2 or NT FAT
  version of encoding software:                   0.3
  minimum file system compatibility required:     MS-DOS, OS/2 or NT FAT
  minimum software version required to extract:   2.0
  compression method:                             deflated
  compression sub-type (deflation):               normal
  file security status:                           not encrypted
  extended local header:                          yes
  file last modified on (DOS date/time):          2018 Jan 2 19:18:10
  32-bit CRC value (hex):                         48c5abda
  compressed size:                                655340 bytes
  uncompressed size:                              1120228 bytes
  length of filename:                             10 characters
  length of extra field:                          28 bytes
  length of file comment:                         0 characters
  disk number on which file begins:               disk 1
  apparent file type:                             binary
  non-MSDOS external file attributes:             000000 hex
  MS-DOS file attributes (00 hex):                none

  The central-directory extra field contains:
  - A subfield with ID 0x000a (PKWARE Win32) and 24 data bytes:
    00 00 00 00 01 00 10 00 00 cd fb 55 28 84 d3 01 00 63 68 25 29 84 d3 01.

  There is no file comment.

```

**Comment by chenxiaolong** [comment]
For reference, this is what minizip 1.2 creates: https://noobdev.io/github/minizip/218/test-1.2.zip

`unzip -t test-1.2.zip` and `7z t test-1.2.zip` do not report errors for this file.

**Comment by nmoinvaz** [comment]
Extra field 0x000a is NTFS timestamp in case you were wondering about that.

**Comment by nmoinvaz** [comment]
It looks like in the test.zip it is not writing the entire compressed content to the zip file (only somewhere around ~655360 (640 * 1024) - instead of 661,246).

**Comment by chenxiaolong** [comment]
Hmm, that's interesting. It seems to do that both with my code and the `minizip` tool.

**Comment by nmoinvaz** [comment]
Ok. I have taken a look. A committed a fix 5bb372d4957ead9aaa56b920a61281fa6ced8cd1 that should solve it.

**Comment by chenxiaolong** [comment]
@nmoinvaz Thank you for the quick fix! Everything is looking good with commit 5bb372d4957ead9aaa56b920a61281fa6ced8cd1.
